### PR TITLE
Trace optimized capacities in ilopf

### LINKED ISSUE #129: track updated impedances in pypsa.linof.ilopf
As discussed in https://github.com/PyPSA/pypsa-eur/pull/116, it would be nice to trace back how the impedances are updated during the iterative optimisation in `ilopf()`. This feature was implemented in pypsa-eur, but won't be available in near future due to the upcoming optimisation without `pyomo` .

### PR DESCRIPTION
**Pull request by FabianHofmann**
Closes #129 

## Changes proposed in this Pull Request
See issue. 
Small modification: the iteration now starts with 1 (started with 0 before). The original capacity is stored under iteration 0. 

## Checklist

- [x] Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.

### CODE REVIEW

File: pypsa/linopf.py [hunk 1]
```diff
@@ -948,6 +948,10 @@ def ilopf(n, snapshots=None, msq_threshold=0.05, min_iterations=1,
     max_iterations : integer, default 100
         Maximal numbder of iterations to run regardless whether msq_threshold
         is already undercut
+    track_iterations: bool, default False
+        If True, the intermediate branch capacity steps and values of the
```
Comment (fneum) [inline pypsa/linopf.py]:
```suggestion
        If True, the intermediate branch capacities and values of the
```

File: pypsa/linopf.py [hunk 2]
```diff
@@ -948,6 +948,10 @@ def ilopf(n, snapshots=None, msq_threshold=0.05, min_iterations=1,
...
+        objective function are recorded for each iteration. The values of
+        iteration 0 stand for the starting point.
```
Comment (fneum) [inline pypsa/linopf.py]:
```suggestion
        iteration 0 represent the initial state.
```

File: pypsa/linopf.py [hunk 3]
```diff
@@ -977,18 +981,33 @@ def msq_diff(n, s_nom_prev):
...
+        if iteration > max_iterations:
             logger.info(f'Iteration {iteration} beyond max_iterations '
                         f'{max_iterations}. Stopping ...')
             break
 
         s_nom_prev = n.lines.s_nom_opt if iteration else n.lines.s_nom
         kwargs['warmstart'] = bool(iteration and ('basis_fn' in n.__dir__()))
-        network_lopf(n, snapshots, **kwargs)
+        status, termination_condition = network_lopf(n, snapshots, **kwargs)
+        assert status == 'ok', ('Optimization failed with status {status}'
```
Comment (fneum) [inline pypsa/linopf.py]:
```suggestion
        assert status == 'ok', (f'Optimization failed with status {status}'
```

File: pypsa/linopf.py [hunk 4]
```diff
@@ -977,18 +981,33 @@ def msq_diff(n, s_nom_prev):
...
+                                'and termination {termination_condition}')
```
Comment (fneum) [inline pypsa/linopf.py]:
```suggestion
                                f'and termination {termination_condition}')
```
Isn't this necessary?

**Comment by FabianHofmann** [comment]
Great, thanks for reviewing!

**Pull request merged**